### PR TITLE
rewrote the subset selector for sgrid, added test based on example notebook

### PR DESCRIPTION
**Pull request by jay-hennen**

### CODE REVIEW

**Review by ChrisBarker-NOAA — CHANGES_REQUESTED** [review]
we need to clean up the tests and examples before merging.

And there should be a notebook example -- is there one already? hopefully with some graphics. See: xarray_subset_grid/visualization/mpl_plotting.py

We should add somenthing there for SGRIDS.




See:

File: tests/test_grids/test_sgrid.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import xarray as xr
+
+import xarray_subset_grid.accessor  # noqa: F401
+from xarray_subset_grid.utils import ray_tracing_numpy
+# open dataset as zarr object using fsspec reference file system and xarray
+def test_polygon_subset():
+    '''
+    This is a basic integration test for the subsetting of a ROMS sgrid dataset using a polygon.
+    '''
+    fs = fsspec.filesystem(
```
[the remark this answers is not in the document]
Comment (jay-hennen) [reply tests/test_grids/test_sgrid.py:13]: I added the arakawa_c_test_grid from gridded for future tests. Will add another offline test soon

[the remark this answers is not in the document]
Comment (ChrisBarker-NOAA) [reply tests/test_grids/test_sgrid.py:13]: thanks!

File: tests/utilities.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+try:
+    import urllib.request as urllib_request  # for python 3
+except ImportError:
+    import urllib2 as urllib_request  # for python 2
+
+import pytest
+import glob
+
+from .get_remote_data import get_datafile
```
Comment (ChrisBarker-NOAA) [inline tests/utilities.py:16]: we don't want to put this in xarray_subset_grid.

I imagine we will put something like this in libgoods, or maybe gridded, but not here.

File: tests/utilities.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+
+def get_test_file_dir():
```
Comment (ChrisBarker-NOAA) [inline tests/utilities.py:19]: we can just put this in the test file directly -- or *maybe* make it a fixture.

And pathlib is great:

from pathlib import Path

`test_file_dir = Path(__file__).parent /  `'test_data')`

File: tests/test_grids/test_sgrid.py [hunk 4]
```diff
@@ -51,4 +59,18 @@ def test_polygon_subset():
...
+    polygon = np.array(
+        [6.5, 37.5],
+        [6.5, 39.5],
+        [9.5, 40.5],
+        [8.5, 37.5],
+        [6.5, 37.5]
+    )
+    ds_subset = ds.xsg.subset_polygon(polygon)
+    
+    breakpoint()
```
Comment (ChrisBarker-NOAA) [inline tests/test_grids/test_sgrid.py:75]: we don't want this in there :-)

File: tests/test_grids/test_sgrid.py [hunk 5]
```diff
@@ -6,6 +6,14 @@
 import xarray_subset_grid.accessor  # noqa: F401
 from xarray_subset_grid.utils import ray_tracing_numpy
 # open dataset as zarr object using fsspec reference file system and xarray
+
+from xarray_subset_grid.utils import get_test_file_dir
+
+from gridded import Variable, VectorVariable
+
```
Comment (ChrisBarker-NOAA) [inline tests/test_grids/test_sgrid.py:13]: I tend to put this inline:

`TEST_DIR = Path(__file__).parent.parent / 'test_data'`

`sample_sgrid_file = TEST_DIR / 'arakawa_c_test_grid.nc'`

**Comment by ChrisBarker-NOAA** [comment]
are the tests running / passing in your repo? 

Check the "actions" tab in the project page.

If not -- I wonder why not?

**Comment by ChrisBarker-NOAA** [comment]
Looks good--- thanks!

I'd love to see more examples and tests, and plotting, but for now this is an improvement.

**Pull request merged**